virtwho/configure.py

```diff
@@ -1,4 +1,119 @@
-"""
-Used to manage virt-who configure file
+"""Used to manage virt-who configure file
 Add/Delete/Update options for /etc/virt-who.conf, /etc/virt-who.d/xxx.conf
 """
+from virtwho.settings import Configure
+from virtwho.settings import config
+from virtwho.ssh import SSHConnect
+from virtwho.logger import getLogger
+
+logger = getLogger(__name__)
+
+
+class VirtwhoHypervisorConfig:
+    """Able to create and manage /etc/virt-who.d/xxx.conf file when
+    call this class"""
+    def __init__(self, mode, register):
+        """Create virt-who configuration file with basic options.
+         All data come from virtwho.ini file.
+        :param mode: The hypervisor mode.
+            (esx, xen, hyperv, rhevm, libvirt, kubevirt, local)
+        :param register: The subscription server. (rhsm, satellite)
+        """
+        self.remote_ssh = virtwho_ssh_connect(mode)
+        self.local_file = f'/root/{mode}.conf'
+        self.remote_file = f'/etc/virt-who.d/{mode}.conf'
+        self.section = f'virtwho-{mode}'
+        self.cfg = Configure(self.local_file, self.remote_ssh, self.remote_file)
+        if mode == 'local':
+            logger.info("Don't need to configure for local mode")
+        else:
+            self.update('type', mode)
+            if mode == 'esx':
+                self.update('server', config.vcenter.server)
+                self.update('username', config.vcenter.username)
+                self.update('password', config.vcenter.password)
+            elif mode == 'xen':
+                self.update('server', config.xen.server)
+                self.update('username', config.xen.username)
+                self.update('password', config.xen.password)
+            elif mode == 'hyperv':
+                self.update('server', config.hyperv.server)
+                self.update('username', config.hyperv.username)
+                self.update('password', config.hyperv.password)
+            elif mode == 'rhevm':
+                self.update('server', config.rhevm.server)
+                self.update('username', config.rhevm.username)
+                self.update('password', config.rhevm.password)
+            elif mode == 'libvirt':
+                self.update('server', config.libvirt.server)
+                self.update('username', config.libvirt.username)
+                self.update('password', config.libvirt.password)
+            elif mode == 'kubevirt':
+                self.update('kubeconfig', config.mode.config_file)
+            if register == "rhsm":
+                self.update('owner', config.rhsm.default_org)
+            elif register == 'satellite':
+                self.update('owner', config.satellite.default_org)
+            self.update('hypervisor_id', 'hostname')
```

Comment:
How about  deal with this like:
```
        if self.register_type == 'rhsm':
            register = config.rhsm
        if self.register_type == 'satellite':
            register = config.satellite
        if self.mode == 'esx':
            hypervisor = config.vcenter
        ======
        if self.mode == 'local':
            logger.waring("Don't need to configure for local mode")
            return False
        ======
        self.update('type', mode)
        self.update('hypervisor_id', 'hostname')
        self.update('owner', register.default_org)
        self.update('server', hypervisor.server)
        self.update('username', hypervisor.username)
        self.update('password', hypervisor.password)

```

Reply:
It seems that we also need to define the hypervisor for each mode.

```
           if self.mode == 'esx':
                hypervisor = config.vcenter
            elif self.mode == 'xen':
                hypervisor = config.xen
            elif self.mode == 'hyperv':
                hypervisor = config.hyperv
            elif self.mode == 'rhevm':
                hypervisor = config.rhevm
            elif self.mode == 'libvirt':
                hypervisor = config.libvirt
            elif self.mode == 'kubevirt':
                hypervisor = config.kubevirt
```